pallets/duster/src/migration.rs

```diff
@@ -79,3 +79,80 @@ pub mod v1 {
 		);
 	}
 }
+
+pub mod v2 {
+	use super::*;
+	use frame_support::traits::{Get, GetStorageVersion, PalletInfoAccess, StorageVersion, UncheckedOnRuntimeUpgrade};
+	use frame_support::storage_alias;
+	use frame_support::Blake2_128Concat;
+	use frame_support::migrations::VersionedMigration;
+
+	#[storage_alias]
+	type AccountBlacklist<T: Config, P: GetStorageVersion + PalletInfoAccess> = StorageMap<P, Blake2_128Concat, <T as frame_system::Config>::AccountId, (), frame_support::pallet_prelude::OptionQuery>;
+
+	// Private module to hide the migration.
+	mod unversioned {
+		use frame_support::pallet_prelude::{GetStorageVersion, PalletInfoAccess};
+
+		pub struct InnerMigrateV1ToV2<T: crate::Config, P: GetStorageVersion + PalletInfoAccess>(core::marker::PhantomData<(T, P)>);
+	}
+
+	impl<T: Config, P: GetStorageVersion + PalletInfoAccess> UncheckedOnRuntimeUpgrade for unversioned::InnerMigrateV1ToV2<T, P> {
+		fn on_runtime_upgrade() -> Weight {
+			log::info!(
+				target: "runtime::duster",
+				"Running migration to v2 for Duster - renaming AccountBlacklist to AccountWhitelist"
+			);
+
+			let mut reads: u64 = 0;
+			let mut writes: u64 = 0;
+
+			// Iterate over all entries in the old AccountBlacklist storage
+			let blacklisted_accounts: Vec<_> = AccountBlacklist::<T, P>::iter().collect();
+			reads += blacklisted_accounts.len() as u64;
+
+			log::info!(
+				target: "runtime::duster",
+				"Migrating {} accounts from AccountBlacklist to AccountWhitelist",
+				blacklisted_accounts.len()
+			);
+
+			// Insert all entries into the new AccountWhitelist storage
+			for (account_id, _) in &blacklisted_accounts {
+				crate::AccountWhitelist::<T>::insert(account_id, ());
+				writes += 1;
+			}
+
+			// Remove all entries from the old storage
+			let removed_keys = AccountBlacklist::<T, P>::drain().count();
```

Comment:
IIRC this leaves the empty storage there. We want to also "kill" it.

Reply:
AFAIK the storage map itself cannot be killed, only the entries, and the drain does that so I think there is no need for additioinal kill. Also, SotrageMap has no kill method, which might explain it